lib/Conversion/TorchToLinalg/Linear.cpp

```diff
@@ -1373,6 +1374,166 @@ class ConvertAtenConvolutionOp : public OpConversionPattern<AtenConvolutionOp> {
+
+  LogicalResult
+  matchAndRewrite(AtenFftRfftOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value self = adaptor.getSelf();
+
+    int64_t dim;
+    auto dimVal = op.getDim();
+    if (isa<torch::Torch::NoneType>(dimVal.getType())) {
+      dim = -1;
+    } else if (!matchPattern(dimVal, torch::Torch::m_TorchConstantInt(&dim))) {
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: requires dim to be constant");
+    }
+
+    if (!isa<torch::Torch::NoneType>(op.getN().getType())) {
+      return rewriter.notifyMatchFailure(op, "unimplemented: parameter n");
+    }
+
+    if (!isa<torch::Torch::NoneType>(op.getNorm().getType())) {
+      return rewriter.notifyMatchFailure(op, "unimplemented: parameter norm");
+    }
+
+    RankedTensorType inputType =
+        cast<RankedTensorType>(adaptor.getSelf().getType());
+    if (!inputType.hasRank()) {
+      return rewriter.notifyMatchFailure(
+          op, "unsupported: only ranked tensors are supported");
+    }
+    if (!inputType.hasStaticShape() || inputType.getRank() > 2) {
+      return rewriter.notifyMatchFailure(
+          op, "unsupported: only static 1D or 2D FFT is supported");
+    }
+
+    const ArrayRef<int64_t> inputShape = inputType.getShape();
+    dim += dim < 0 ? inputShape.size() : 0;
+
+    const int64_t fftLength = inputShape[dim];
+    const int64_t rank = inputType.getRank();
+    const int64_t lastDim = rank - 1;
+    const int64_t outputFftDim = fftLength / 2 + 1;
+    const bool needTranspose = dim != lastDim;
+
+    RankedTensorType newResultType = llvm::cast<RankedTensorType>(
+        getTypeConverter()->convertType(op.getType()));
+    llvm::SmallVector<int64_t> componentShape(newResultType.getShape());
+
+    // Transpose if FFT dimension is not the last one
+    llvm::SmallVector<int64_t> perms = llvm::to_vector(llvm::seq(rank));
+    std::swap(perms[dim], perms[lastDim]);
+    if (needTranspose) {
+      self = transposeValue(loc, self, perms, rewriter);
+      for (size_t i = 0; i < componentShape.size(); i++) {
+        componentShape[i] = newResultType.getShape()[perms[i]];
+      }
+    }
+
+    RankedTensorType matrixType = RankedTensorType::get(
+        {fftLength, outputFftDim}, inputType.getElementType());
+
+    RankedTensorType componentsType =
+        RankedTensorType::get(componentShape, inputType.getElementType());
+
+    Value realMatrix =
+        getDFTMatmulCoeff(rewriter, loc, matrixType, /*isRealPart=*/true);
+    Value real = createLinalgMatmulOnTensors(rewriter, loc, componentsType,
+                                             self, realMatrix);
+
+    Value imagMatrix =
+        getDFTMatmulCoeff(rewriter, loc, matrixType, /*isRealPart=*/false);
+    Value imag = createLinalgMatmulOnTensors(rewriter, loc, componentsType,
+                                             self, imagMatrix);
```

Comment:
I think my comment about multiple matmuls in the decomposition below applies here as well. Let me know what you think about making one `DFTMatmulCoeff` with both real and imaginary parts.

Reply:
Although, in this case, the linalg generic would need to be constructed more carefully, since you wouldn't have two tensors to iterate over (it wouldn't be elementwise anymore, and you would need to fiddle with the indexing maps). Feel free to keep this conversion as-is if it seems like too much work to make the change.

Reply:
Refactored the conversion in the last commit.